File: Lezioni_1_2_3_4/steleDiRosetta/loop.c
#include <stdio.h>
#include <math.h>
#include <time.h>

int main ()
{
	int NLOOP=100000;
        long p2,p3;
        double sq;
        clock_t start, stop;
        double time_spent;

        printf("Begin loop\n");
        start = clock();
 
	for (int i=0; i<NLOOP; i++){
           p2 = (long) pow((double) i,2);
           p3 = (long) pow((double) i,3);
           sq = sqrt((double) i);
           printf( "%d\t%ld\t%ld\t%5.3lf\n",i,p2,p3,sq); 
        }

        printf("End loop\n");
        stop = clock();
        time_spent = (double)(stop - start) / CLOCKS_PER_SEC;
        printf("Done in %lf seconds\n", time_spent );
   
   return(0);
}

File: Lezioni_1_2_3_4/steleDiRosetta/loop.py
# -*- encoding: utf-8 -*-
#Esempio che mostra il loop e le chiamate a finzioni esterne

import math
import time


NLOOP=1000000;


if __name__ == '__main__':

    print "Begin loop"

    start = time.time()*1000.0

    for i in range(0,NLOOP):
        p2 = long(math.pow(i,2.0))
        p3 = long(math.pow(i,3.0))
        sq = math.sqrt(i)
        print "%d\t%d\t%d\tRADICE:%5.3f"%(i,p2,p3,sq)
        print "-"*80
        #print "ho sonno vado a dormire"
        #time.sleep(3)
        #print "mi sono svegliato"

    print "End loop"
    stop = time.time()*1000.0
    time_spent = stop - start
    print "Done in %3.2f millsec"%time_spent